examples/esp-todomvc-react/js/app.jsx

```diff
@@ -0,0 +1,65 @@
+/*global React, Router*/
+var app = app || {};
+
+(function () {
+    "use strict";
+
+    var Header = app.Header,
+        Main = app.Main,
+        Footer = app.Footer;
+
+    var TodoApp = React.createClass({
+
+        componentDidMount: function () {
+            var router = Router({
+                "/": this.showAll,
+                "/active": this.showActive,
+                "/completed": this.showComplete
+            });
+            router.init("/");
+        },
+
+        showAll: function () {
+            this.props.router.publishEvent("todoList", "filterChanged", { filter: app.model.Filter.all });
+        },
+
+        showActive: function () {
+            this.props.router.publishEvent("todoList", "filterChanged", { filter: app.model.Filter.active });
+        },
+
+        showComplete: function () {
+            this.props.router.publishEvent("todoList", "filterChanged", { filter: app.model.Filter.complete });
+        },
+
+        render: function () {
+            var model = this.props.model,
+                router = this.props.router;
+            return (
+                <div>
+                    <Header router={router} />
+                    <Main model={model.main}
+                          router={router}
```

Comment:
Yeah agreed. Out of interest, do we have a strategy for unit testing context in React?